sources/vega/data/v1/data.proto

```diff
@@ -0,0 +1,32 @@
+syntax = "proto3";
+
+package data.v1;
+
+option go_package = "code.vegaprotocol.io/protos/vega/data/v1";
+
+// ExternalData describes valid external source data that has been received by the node.
+// It represents both matched and unmatched data.
+message ExternalData {
+  // pubKeys is the list of authorized public keys that signed the data for this
+  // source. All the public keys in the data should be contained in these
+  // public keys.
+  repeated string pub_keys = 1;
```

Comment:
Should we make this a oneof?
```
message Signer {
   oneof signer {
      string pub_key = 1;
      string address = 2;
   }
 }

message ExternalData {
    repeated Signer signers = 1;
}
```

The reason is that in some cases e.g: an open oracle, an ethereum address will be submitted, in the case of vega signed oracle it will be a vega pubkey.

Not sure it's required, but would make sense

Reply:
This has been fixed https://github.com/vegaprotocol/protos/pull/514/files#diff-e5814c05abaec52ff84d42ee4896683612257a404b721c65b14efe5febcb762eR7-R31
Thanks!